sdk/eventhub/Azure.Messaging.EventHubs/src/Producer/PublishFailedEventArgs.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+
+namespace Azure.Messaging.EventHubs.Producer
+{
+    /// <summary>
+    /// This class allows the application and the producer to pass contextual information
+    /// back and forth on the event of a batch failing to be sent to the consumer successfully.
+    /// </summary>
+    public class PublishFailedEventArgs : EventArgs
+    {
+        /// <summary>
+        /// Events is an IEnumerable of the events in the EventDataBatch that failed to send.
+        /// </summary>
+        public IEnumerable<EventData> Events { get; set; }
+
+        /// <summary>
+        /// Exception is the Exception that was thrown when the EventDataBatch failed to send.
+        /// </summary>
+        public Exception Exception { get; set; }
+
+        /// <summary>
+        /// PartitionId is the id of the partition that the EventDataBatch was being sent to when it
+        /// failed.
+        /// </summary>
+        public int PartitionId { get; set; }
```

Comment:
```suggestion
        public string PartitionId { get; set; }
```

The service doesn't guarantee us that we can safely cast this.